### ISSUE TITLE
Support for multiresolution (pyramid) layers besides Image layers

### ISSUE THREAD

**Issue by GenevieveBuckley** [napari/napari#1019]
## 🚀 Feature

1. Move everything needed for multiresolution/pyramid layers into VispyBaseLayer. Currently we have things split across VispyImageLayer and VispyBaseLayer, eg: the `find_top_left()` method, which depends on the `level_shapes` property (which is only available in the image layer).
2. (Possibly) change the name of `is_pyramid` to `is_multiresolution`.

## Motivation

The motivation here is that we'd be in a better position to support multiresolution display for layer types besides Image layers. 
Eg: points layers with *lots* of points maybe don't need all of them rendered if you're zoomed out a lot, etc.

## Pitch

This isn't a new pitch, it's more documentation of an idea that's been floating around for a while.

https://github.com/napari/napari/pull/1015#issuecomment-593254516 by @jni

> I don't remember where this concept came up last week, but we had the idea that `is_pyramid` would become `is_multiscale` and would apply across all layers (eg millions of points would get grouped into a single point if you zoom out enough, just like this happens with some mapping applications, for example). So, I think a lot of that logic should definitely go on the base layer.

**Comment by sofroniewn** [comment]
Thanks for making this issue!! I like this idea, it would maybe be good to have a concrete example of say many many points where napari currently cannot handle it, but where having a multiresolution approach would help. Maybe @kevinyamauchi has ideas / experience there

**Comment by d-v-b** [comment]
This seems like a good idea; additionally, it might be nice to force all data to be represented internally as multiscale, even if only 1 scale level (full resolution) is used / calculated. This would remove a lot of the current `is_pyramid` checks.

**Comment by sofroniewn** [comment]
Noting the discussion in https://github.com/zarr-developers/zarr-specs/issues/50 it looks like their preferred term would be `is_multiscale` and I see that is what @jni used above. Do we prefer that or `is_multiresolution` which I think I had used (without much thought)? I'm fine with either but would like us to match them if we can.

**Comment by d-v-b** [comment]
to be super pedantic, "multiscale" is more implementation-agnostic than "multiresolution". The resolution of actual images can be decoupled from the sampling density -- two images can have the same resolution, but different sampling densities. "multiscale" more clearly describes the space in which the image is embedded. Also multiscale is a little shorter :)

**Comment by sofroniewn** [comment]
I'd like to do the `is_multiscale` switch in my next PR in the world coordinates series (after #1042 merges). Given the amount of code that depends on `is_pyramid` and our plans to make our next release `0.3.0` we could either add a deprecation warning or maybe catch the error and give a special update message. We can decide in the context of that PR but curious what @jni and others think

**Comment by joshmoore** [comment]
In case any of this happens to half-way useful:

> https://github.com/napari/napari/issues/1019#issuecomment-594191704 it might be nice to force all data to be represented internally as multiscale, even if only 1 scale level

"Everything has at least resolution 0" is what we've decided to go with in https://github.com/ome/omero-ms-zarr/blob/master/spec.md

We have images converted via [bioformats2raw](https://github.com/glencoesoftware/bioformats2raw) in public S3 if anyone needs them, and an upcoming sprint will look into doing the same with masks.

**Comment by sofroniewn** [comment]
Thanks for those links!

> "Everything has at least resolution 0" is what we've decided to go with in https://github.com/ome/omero-ms-zarr/blob/master/spec.md

We might adopt that internally then

**Comment by psobolewskiPhD** [comment]
So layers now use `.multiscale` and it's implemented for both image and labels (Class ScalarField).
Does `multiscale` make sense for the other layers?
cc: @jni

**Comment by jni** [comment]
Yes, you can find multiscale points in various settings like maps where multiple points overlap and they get coalesced into a (bigger) spot with a points count. For surfaces, you have have surfaces with varying level-of-detail and only load the detailed meshes when you zoom into an object. Similarly for vector fields, etc.

**Comment by kephale** [comment]
We're working on multiscale meshes in ngff: https://github.com/ome/ngff/issues/33

I have code for reading multiscale meshes in napari: https://github.com/kephale/atrium.kyleharrington.com/blob/main/meshes/view_in_napari/demo.py

However, I'm not doing anything smart it just opens all meshes at all scales. I figured folks might be interested or would have opinions before I get too carried away 😅
